### ISSUE TITLE
Stop hiding posts with "Load More"

### ISSUE THREAD

**Issue by githupper** [tuskyapp/Tusky#3461]
* * * *
- Tusky Version: 21.0
- Android Version:
- Android Device:
- Mastodon instance (if applicable):

Whenever I resume browsing from Mastodon, usually there would be new posts but Tusky always hide some posts under "Load More" and some posts above it. 

So if I want to read those hidden posts, I would need to tap "Load More" them scroll below to read those posts. But once I'm done reading those posts, I had to scroll all the way to view those posts previously above "Load More".

For example I have these posts
1
2
3
4
5
6
7
8.

Posts 4 to 8 are hidden behind " Load More". To view posts 4 to 8 I have to tap "Load More" then scroll down to 8. However once I done that, I haven't read post 1 to 3 so I  have to scroll from 8 to 4 again then I can continue to read posts 3 2 1 which doesn't make sense .

Hence I would like to request doing away with "Load More" and show all posts every time.

Thanks

**Comment by Lakoja** [comment]
(just to clarify)

In your example when you resume you start out at status 9? And then you would want to read 8 up to 1?

(so "oldest first")

**Comment by githupper** [comment]
Hi,

What I meant was that in the example I gave, when I launched the app and there were 10 new posts, some of the posts were hidden under "Show More" while some are not. 

At this point, I can choose to scroll up and see Post 4th, 3rd, 2nd, and 1st. If I am interested to know what are the other posts hidden under Show More, I would need to scroll 1,2,3,4, tap Show More, then scroll down 5,6,7, and 8. 9th post would be where I last close the app.

If I had chosen to tap Show More instead and read, 5th, 6th, 7th, 8th, when I reached the 9th post, that is again my last viewed post before I close the app. So in order to read those posts above 5th, I would need to scroll 8th, 7th, 6th, 5th, 4th, 3rd, 2nd then 1st post.

I know my explanation might be hard to understand but I just want to have an option to not having to tap Show More to view all new posts. Because doing that forces the user to choose to scroll up then down to view every post or scroll down first then scroll all the way to the top to read all new posts.

The official Mastodon app which I am using, just have a See new posts alert and I can scroll up to view every single new posts.

**Comment by timfoster** [comment]
This is something I'd quite like as well, and while I'm  not sure I completely understand the original reporter's intent, I suspect we're talking about the same thing.

To clarify: no need to actually disable the "load more" button, just change how we set the currently-reading bookmark in the app after hitting "load more".

I always read, bottom to top, posts in chronological order. Today, after hitting "Load more", Tusky sets the post pointer to the newest entry in that expanded list of "load more" posts, rather than either exactly where it was when I hit the button, or at the start of the last (oldest) post in the expanded "load more" list.

Does that describe the intent of this issue, or should I open a new issue for my request?

Thanks for all the hard work on Tusky, it's still the best Android app for Mastodon, and I really appreciate it!

**Comment by Tak** [comment]
> Today, after hitting "Load more", Tusky sets the post pointer to the newest entry in that expanded list of "load more" posts, rather than either exactly where it was when I hit the button, or at the start of the last (oldest) post in the expanded "load more" list.

Have you set the `Reading Order` preference to `Oldest first`?

**Comment by nikclayton** [comment]
I've got very rough code that eliminates "Load more".

https://youtube.com/shorts/2kboOYH1olU?feature=share shows what it looks like, starting with an empty timeline and scrolling through.

**Comment by timfoster** [comment]
> Have you set the Reading Order preference to Oldest first?

Nope, I leave it on "newest first". On launching, the application opens on the last read post, and we scroll "upwards" to view newer posts. On encountering a "Load more" button, the assumption is that it expands the list of newer posts upwards, leaving our current view at the post immediately below the now-expanded list.

If it's been a long time since reading, the user may choose not to expand the list of posts, and continue scrolling up to see newer posts.

I'll experiment with oldest first to see whether that makes more sense - it'll probably take a bit of adjustment to get my brain around it :-)

**Comment by nikclayton** [comment]
> On encountering a "Load more" button, the assumption is that it expands the list of newer posts upwards, leaving our current view at the post immediately below the now-expanded list.

That's *exactly* what changing the reading order to "Oldest first" will do for you.

Nothing else changes when you set that option.

**Comment by githupper** [comment]
Now that I explore other Mastodon clients, I realized every single one of them has this issue. Mastodon itself, Megalodon, Tusky, list goes on. I've also checked, every 20 posts or so (at least in main Mastodon client" it will show me the dreaded  "Show More", "Load More", "Load missing posts" just like other clients. Why?!

I'm not sure what all these clients are trying to do by asking users but it is very bad design language. Not to mention all the Mastodon devs copying the same issue. It is frustrating enough to make me stop using Mastodon on a regular basis.

I think this thread can close as I'm done with Mastodon. For anyone who is keen, please continue or make a new thread if needed. I'm not commenting on this issue here moving forward. 

I appreciate the devs for working on Mastodon but I stick to other sources for my open source news. Thanks

**Comment by nikclayton** [comment]
@githupper 

1. Change your "Preferences > Reading order" to oldest first. That will remove the scrolling that you have to do.
2. As I mentioned in https://github.com/tuskyapp/Tusky/issues/3461#issuecomment-1509209470, I am actively working on removing "Load more" entirely from Tusky. That won't be in v22, but I hope to have it ready for v23.

**Comment by Fell** [comment]
I support removing "Load more" for other reasons.

I often find myself missing the "Load more" button and reading posts from one or two days ago until I notice something I have seen before. Then I have to scroll back to find the "Load more" button I missed. Depending on my screen brightness, it can be hard to see.

I would prefer the endless scrolling like the Mastodon web app already does, and it seems like @nikclayton is already working on that (thanks!).

Some people might see the button as a digital well-being feature, offering a time to stop and preventing them from actually scrolling endlessly. To make up for that, I propose an optional feature that inserts a card with a text like "You have seen 25 posts today." But that would be a different issue.

**Comment by edcarlo157** [comment]
I also would like this to removed or at least be optional. I often notice that, even after expanding the "load more" button sometimes 1 day+ old posts are skipped entirely. 
Using the example of the first post: I'd start on 8 and the posts 1-7 would be hidden under "load more", after expanding it, tusky loads 1,2,3,4,8 without any "load more" or anything at all in between 4 and 8, I would know something is missing by checking the time of posting of 4 and 8, often there is 8+ hours in between them
